### PR TITLE
WOCOperationMode should change UIUserInterfaceIdiom. Support UIModalPresentationFormSheet on tablets (non-fullscreen presentation)

### PR DESCRIPTION
**Pull request by ehren**
Also adds examples to WOCCatalog app including resizing modal via
preferredContentSize and toggling tablet mode at runtime.

-----

First, sorry for going AWOL on the last patch!

A couple of things:
This patch doesn't address blocking touches on the parent's view (doesn't even set userInteractionEnabled = FALSE). I'm thinking that should be handled by a dummy full screen view underneath the modal - the same view could be used for the gray overlay applied to the parent's view when using a popover.

I require iPad style form sheet modals and popovers in my app. Are there any unreleased changes regarding popover support pending? I'm thinking popover support is not that bad e.g. not all that different from form sheet (I believe on iOS form sheet modals and popovers are presented in separate UIWindows which apparently doesn't need to be the case here). So the positioning logic for popovers would be similar (though ideally not embedded into UIViewController.mm also!). Popovers would also require the same "if not iPad use full screen" logic.

### CODE REVIEW

**Comment by Ramu-msft** [comment]
@davelamb is added to the review. #Closed

**Comment by Ramu-msft** [comment]
@bdrlamb-ms is added to the review. #Closed

File: Frameworks/UIKit/UIViewController.mm [hunk 1]
```diff
@@ -384,6 +377,35 @@ - (CGRect)_orientationRect:(UIInterfaceOrientation)orientation {
...
 }
 
+- (CGRect)_modalPresentationFormSheetFrame {
+    if (!GetCACompositor()->isTablet()) {
+        // fullscreen on non-tablets
+        return [[UIScreen mainScreen] applicationFrame];
+    }
+
+    CGRect frame;
+    if (!CGSizeEqualToSize(priv->_preferredContentSize, CGSizeZero)) {
```
Comment (davelamb) [inline Frameworks/UIKit/UIViewController.mm]: Should use accessors for these.

File: Frameworks/UIKit/UIViewController.mm [hunk 2]
```diff
@@ -342,14 +342,7 @@ - (CGRect)_orientationRect:(UIInterfaceOrientation)orientation {
     appFrame = [[UIScreen mainScreen] applicationFrame];
 
     if (priv->_presentationStyle == UIModalPresentationFormSheet) {
```
Comment (davelamb) [inline Frameworks/UIKit/UIViewController.mm]: Not part of the change, but again should be using accessors for accessing properties.

**Comment by bdrlamb** [comment]
:+1:

**Comment by ehren** [comment]
I wasn't exhaustive with the accessors switch in the last commit - e.g. didn't touch priv->_view and some instances of private access to _parentViewController even in non init contexts seemed better to leave private. Let me know if that went overboard though (suppose there's a bit of overhead in the objective-c message dispatch added here).

I'm going to put further comments related to popovers in issue #302.

**Comment by ehren** [comment]
No rush on these changes but perhaps I misunderstood the comment i.e. I should instead change UIViewControllerPriv to a class with public accessors?

In any case, I'm not sure if the public properties of e.g. UIViewController are invoked internally on iOS - you're definitely going to have a bad time with any non-trivial overriding of e.g. -(UIViewController*)parentViewController; but that's probably the case on iOS too.

**Comment by davelamb** [comment]
Anything private can be accessed however you want (a struct is fine), I'm just worried about public-facing properties, since the accessors can be overridden. If we poke at the backing ivars outside the accessors, strange behaviors can pop up. This is just a general rule we haven't been too good about following, so I figured while you're in there... :)

Thanks!

---
In reply to: [212905371](https://github.com/Microsoft/WinObjC/pull/450#issuecomment-212905371) [](ancestors = 212905371)

**Comment by davelamb** [comment]
:+1:
